### ISSUE TITLE
Remove unused code

### ISSUE THREAD

**Issue by Afforess** [Afforess/Factorio-Stdlib#105]
I have a sneaking suspicion that 10-20% of Factorio-Stdlib is not used by any modders today. It may be because parts of the API are not particularly useful (`table.each`) or because they are too complicated to use properly/correctly (`trains`). 

One way to find out would be to write a script to login to the mod portal, download every mod published, and unzip and check for the stdlib, and any of the functions within. It should be possible to scrap together an idea of the number of used functions and which no one cares about.

**Comment by Nexela** [comment]
I use table.each and table.map and table.arr_to_bool and just about everything else :)))

Regarding some of the stuff I don't use
(trains) but only because it needs more fleshing out. 

(tile/chunk)  because I don't know if I trust the storing/retrieving and havn't spent a lot of time looking at it.  If anything the non saving stuff should be moved to position/area and the storing/getting should be left in.

Event -> I use the heck out of this even though Rseding hates on it :)
Event.Gui -> because just awesome
Event.Time -> I don't have any mods that need to keep track of day events but it def usuable!
Event.Player, Event.Force -> Well I added these sooooo that should tell you something :)
Event.Configuration -> When this is done (after 1.0) will make managing config_changed stuff easy

table and string, I use depending on the situation,   
config -> Torn on this one,  I don't use it for managing global table stuff. but I do use it for managing config stuff
Logger -> Needs some tweaks but I use it occasionally
Entity -> the new build/destroy/revive stuff is neat,  the rest I don't use
Surface -> Don't use it all though I can see some of its uses

**Comment by trappi-stone** [comment]
@Afforess @Nexela I thought it was a good idea, so I picked it up and worked on the script. I'm not entirely confident about my script's accuracy, but it seems to work as I intended so far.

Link to the scripts: https://github.com/trappi-stone/factorio-stdlib-stat
Link to the results: https://docs.google.com/spreadsheets/d/1UDVG8CDIlSDUe51nc427w2H9e_-ihFXVCuFRtBCni5A/pubhtml

**Things to point out about the script:**

- The script expects the stdlib folder to exist at the top-level of the mod directory.
- For now, the script uses stdlib ***v0.8.0*** to check whether or not a mod uses the functions from it.
- The script parses all the functions defined in stdlib.
- The script parses as many function invocations as possible from a target mod.
- Compares if the mod is invoking a function that exists in stdlib.

**Description of the results:**

- The header is a list of all the functions defined in stdlib ***v0.8.0***.
- The far left column is a list of all the mods that have **stdlib** folder at the top-level of the mod.
- 0 indicates that a mod uses the corresponding function.
- 1 indicates that a mod uses the corresponding function.
- The last row shows the sum of all the function usage across all mods that use stdlib.

**Several guesses about the results:**

- There might be mods that use the modules from stdlib and yet its modder might have placed them in a folder other than **stdlib**.
- There might be mods that use the functions from stdlib and yet the modder might have renamed it.
- There might be mods that use an older version of stdlib (not 0.8.0).

**Comment by Nexela** [comment]
I think the biggest issues are the documentation for the functions.  trappi-stone has done great work making them better, however some functions need better explanations specifically the table. stuff.

When I was first learning Lua a lot of the table. stuff seemed like gobbly gook and the examples were hard to follow.

**Comment by trappi-stone** [comment]
@Afforess 
> I have a sneaking suspicion that 10-20% of Factorio-Stdlib is not used by any modders today.

Based on the analysis across 49 mods that have the `stdlib` folder, the function usage ranges from 0% to 24% for stdlib v0.8.0.

@Afforess @Nexela 
> It may be because parts of the API are not particularly useful (table.each) or ... 

It's true that a mod might not need a particular stdlib module over another &mdash; entirely depends on the type of mod that a modder is building. The same modder might find a different module useful if they decide to create a new mod that can make use of such module.

> ... because they are too complicated to use properly/correctly (trains).

> I think the biggest issues are the documentation for the functions.

> When I was first learning Lua a lot of the table. stuff seemed like gobbly gook and the examples were hard to follow.


I agree that the documentation still needs a lot of work, to the point where a beginning modder doesn't even have to peek at the code to understand what a function does and in what context it is used.

For the current version of stdlib (master), I went through every single module in stdlib and addressed ldoc formatting issues, re-worded some existing descriptions, and added more descriptions for several functions to clarify their functionality.

Nevertheless, the documentation still has a HUGE room for improvement.
